pandas/core/indexes/multi.py

```diff
@@ -1412,6 +1412,45 @@ def from_product(cls, iterables, sortorder=None, names=None):
         labels = cartesian_product(labels)
         return MultiIndex(levels, labels, sortorder=sortorder, names=names)
 
+    @classmethod
+    def from_frame(cls, df, squeeze=True):
+        """
+        Make a MultiIndex from a dataframe
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            DataFrame to be converted to MultiIndex
+        squeeze : bool
+            If df is a single column, squeeze multiindex to be a regular index.
+
+        Returns
+        -------
+        index : MultiIndex
```

Comment:
no need for giving a name, just leave the type. Add a description in the next line (indented).